iac-common/src/testFixtures/java/org/sonar/iac/common/testing/IacTestUtils.java

```diff
@@ -92,4 +92,14 @@ public static InputFileContext createInputFileContextMock(String filename, Strin
     when(inputFile.language()).thenReturn(languageKey);
     return inputFileContext;
   }
+
+  public static InputFileContext createInputFileContextMockFromContent(String content, String filename, String languageKey) {
+    var inputFileContext = createInputFileContextMock(filename, languageKey);
+    try {
+      when(inputFileContext.inputFile.contents()).thenReturn(content);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    return inputFileContext;
+  }
 }
```

Comment:
Please provide a good description to this yet another mocking method. I feel we actually loose control over all the test helper method to mock something.

Reply:
With my change the `InputFileContex` passed to `TreeParser.parse(String source, @Nullable InputFileContext inputFileContext)` has to be fully mocked (i.e. `InputFileContex.inputFile.content()` returns a real value).
Before, we didn't need that because `ArmTemplateExpressionParser` didn't need access to the real course code. 

The alternative solution is to store file content in the `ArmJsonBaseConverter` field (and pass it to many places that will generate more changes.

Reply:
But it's a valid point - another method should be replaced by this one.